subsys/bluetooth/host/addr.c

```diff
@@ -120,3 +120,19 @@ bool bt_addr_le_is_resolved(const bt_addr_le_t *addr)
 {
 	return (addr->type & ADDR_RESOLVED_BITMASK) != 0;
 }
+
+bool bt_addr_le_is_valid(const bt_addr_le_t *addr)
+{
+	if (addr == NULL) {
+		return false;
+	}
+
+	if (addr->type != BT_ADDR_LE_PUBLIC && addr->type != BT_ADDR_LE_RANDOM &&
+	    addr->type != BT_ADDR_LE_PUBLIC_ID && addr->type != BT_ADDR_LE_RANDOM_ID) {
+		return false;
+	}
+
+	/* Additional checks for validity can be added here based on the core spec */
+
```

Comment:
IIRC there are some bits of the address that needs to be set/unset in the case of NRPAs and RPA (or something like that). I'll see if I can dig anything specific up

Reply:
We have a macro for checking random addresses here: https://github.com/zephyrproject-rtos/zephyr/blob/e4747b28a8f3edac3ba8282bb7172898940bcd37/include/zephyr/bluetooth/addr.h#L142-L148

The complete check for random is `type != BT_ADDR_LE_RANDOM || BT_ADDR_IS_RPA(a) || BT_ADDR_IS_NRPA(a) || BT_ADDR_IS_STATIC(a)`.

I think the 'public' namespace is EUI-48. Not sure. But if so, that also has some requirements.

Reply:
> We have a macro for checking random addresses here:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/e4747b28a8f3edac3ba8282bb7172898940bcd37/include/zephyr/bluetooth/addr.h#L142-L148
> 
> The complete check for random is `type != BT_ADDR_LE_RANDOM || BT_ADDR_IS_RPA(a) || BT_ADDR_IS_NRPA(a) || BT_ADDR_IS_STATIC(a)`.
> 
> I think the 'public' namespace is EUI-48. Not sure. But if so, that also has some requirements.

As per the issue (https://github.com/zephyrproject-rtos/zephyr/issues/71801) there are additional requirements for random addresses, such as at least one bit of the random part shall be 0 and one shall be 1. 

For example 40:00:00:00:00:00 and 40:ff:ff:ff:ff:ff are not valid RPAs, but will `BT_ADDR_IS_RPA` will return `true` on those